src/Orchard.Web/Modules/Orchard.AuditTrail/Views/AuditTrailEvent-Content.SummaryAdmin.cshtml

```diff
@@ -1,37 +1,34 @@
-@using Orchard.AuditTrail.Helpers
+﻿@using Orchard.AuditTrail.Helpers
 @using Orchard.AuditTrail.Services.Models
 @using Orchard.ContentManagement
 @{
     var descriptor = (AuditTrailEventDescriptor)Model.Descriptor;
     var eventData = (IDictionary<string, object>) Model.EventData;
     var eventPastTense = descriptor.Name.Text.ToLower();
-    var contentItemId = (int?)Model.ContentItemId;
+    var contentItemId = (int)Model.ContentItemId;
     var contentItem = (ContentItem) Model.ContentItem;
     var eventVersionNumber = eventData.Get<int>("VersionNumber");
     var isPublishedEvent = eventData.Get<bool>("Published");
-    var itemDisplayText = contentItem != null ? Html.ItemDisplayText(contentItem) : default(IHtmlString);
-    var title = itemDisplayText != null ? itemDisplayText.ToString() : eventData.Get<string>("Title");
-    if (string.IsNullOrEmpty(title))
-    {
-        title = "[untitled]";
-    }
+    var itemDisplayText = contentItem != null ? Html.ItemDisplayText(contentItem, false) : default(IHtmlString);
```

Comment:
This is wrong, because you are not encoding the value, and it's correctly encoded only if it's rendered by ItemEditLink lower. But there is a case where it's passed to `Html.Raw`, so in this case it needs to be encoded.

The solution is to not use ItemEditLink but ItemEditUrl and build the <a> yourself.